src/sinks/influxdb/mod.rs

```diff
@@ -0,0 +1,778 @@
+) {
+    // Fields
+    let unwrapped_fields = fields.unwrap_or_else(|| HashMap::new());
+    // LineProtocol should have a field
+    if unwrapped_fields.is_empty() {
+        return;
+    }
+
+    encode_string(measurement, line_protocol);
+    line_protocol.push(',');
+
+    // Tags
+    let mut unwrapped_tags = tags.unwrap_or_else(|| BTreeMap::new());
+    unwrapped_tags.insert("metric_type".to_owned(), metric_type.to_owned());
+    encode_tags(unwrapped_tags, line_protocol);
+    line_protocol.push(' ');
+
+    // Fields
+    encode_fields(unwrapped_fields, line_protocol);
+    line_protocol.push(' ');
+
+    // Timestamp
+    line_protocol.push_str(&timestamp.to_string());
+    line_protocol.push('\n');
+}
+
+fn encode_tags(tags: BTreeMap<String, String>, output: &mut String) {
+    let sorted = tags
+        // sort by key
+        .iter()
+        .collect::<BTreeMap<_, _>>();
+
+    for (key, value) in sorted {
+        if key.is_empty() || value.is_empty() {
+            continue;
+        }
+        encode_string(key.to_string(), output);
+        output.push('=');
+        encode_string(value.to_string(), output);
+        output.push(',');
+    }
+
+    // remove last ','
+    output.pop();
+}
+
+fn encode_fields(fields: HashMap<String, Field>, output: &mut String) {
+    for (key, value) in fields.into_iter() {
+        encode_string(key.to_string(), output);
+        output.push('=');
+        match value {
+            Field::String(s) => {
+                output.push('"');
+                for c in s.chars() {
+                    if "\\\"".contains(c) {
+                        output.push('\\');
+                    }
+                    output.push(c);
+                }
+                output.push('"');
+            }
+            Field::Float(f) => output.push_str(&f.to_string()),
+            Field::UnsignedInt(i) => {
+                output.push_str(&i.to_string());
+                output.push('u');
+            }
+            Field::Int(i) => {
+                output.push_str(&i.to_string());
+                output.push('i');
+            }
+            Field::Bool(b) => {
+                output.push_str(&b.to_string());
+            }
+        };
+        output.push(',');
+    }
+
+    // remove last ','
+    output.pop();
+}
+
+fn encode_string(key: String, output: &mut String) {
+    for c in key.chars() {
+        if "\\, =".contains(c) {
+            output.push('\\');
+        }
+        output.push(c);
+    }
+}
+
+fn encode_timestamp(timestamp: Option<DateTime<Utc>>) -> i64 {
+    if let Some(ts) = timestamp {
+        ts.timestamp_nanos()
+    } else {
+        encode_timestamp(Some(Utc::now()))
+    }
+}
+
+fn encode_namespace(namespace: &str, name: &str) -> String {
+    if !namespace.is_empty() {
+        format!("{}.{}", namespace, name)
+    } else {
+        name.to_string()
+    }
+}
+
+fn encode_uri(
+    endpoint: &str,
+    path: &str,
+    pairs: &mut [(&str, Option<String>)],
```

Comment:
I think this was present in the original source, but it really bothers me. I don't see why this needs to be `&mut` at all (and so requiring the same in all the callers). The pairs are not modified in this function that I can see.

Reply:
Yeah, you are right. I just fixed it.